core-api/src/main/java/com/optimizely/ab/config/audience/UserAttribute.java

```diff
@@ -75,66 +81,55 @@ public Object getValue() {
+        // TODO(llinn) don't create inner condition every evaluation
+        Condition<?> delegate = getCondition();
+        try {
+            return delegate.evaluate(config, attributes);
+        } catch (Exception e) {
+            logger.error("Evaluation failed for '{}' match condition on '{}' attribute", match, name, e);
         }
-        // Valid for primitive types, but needs to change when a value is an object or an array
-        Object userAttributeValue = attributes.get(name);
+        return null;
+    }
 
-        if (!"custom_attribute".equals(type)) {
-            logger.warn("Audience condition \"{}\" has an unknown condition type. You may need to upgrade to a newer release of the Optimizely SDK", this);
-            return null; // unknown type
-        }
-        // check user attribute value is equal
+    /**
+     * Provides a {@link Condition} that implements match-specific behavior.
+     *
+     * If this object does hold a valid configuration for a leaf condition,
+     * i.e. an unknown {@link #match} type or {@link #value} that is not compatible
+     * with {@link #match}, this will return a {@link Condition} that always evaluates
+     * to UNKNOWN.
+     */
+    private Condition<?> getCondition() {
+        Condition<?> condition;
         try {
-            Match matchType = MatchType.getMatchType(match, value).getMatcher();
-            Boolean result = matchType.eval(userAttributeValue);
-
-            if (result == null) {
-                if (!attributes.containsKey(name)) {
-                    //Missing attribute value
-                    logger.debug("Audience condition \"{}\" evaluated to UNKNOWN because no value was passed for user attribute \"{}\"", this, name);
-                } else {
-                    //if attribute value is not valid
-                    if (userAttributeValue != null) {
-                        logger.warn(
-                            "Audience condition \"{}\" evaluated to UNKNOWN because a value of type \"{}\" was passed for user attribute \"{}\"",
-                            this,
-                            userAttributeValue.getClass().getCanonicalName(),
-                            name);
-                    } else {
-                        logger.warn(
-                            "Audience condition \"{}\" evaluated to UNKNOWN because a null value was passed for user attribute \"{}\"",
-                            this,
-                            name);
-                    }
-                }
-            }
-            return result;
-        } catch (UnknownMatchTypeException | UnexpectedValueTypeException ex) {
-            logger.warn("Audience condition \"{}\" " + ex.getMessage(),
-                this);
-        } catch (NullPointerException np) {
-            logger.error("attribute or value null for match {}", match != null ? match : "legacy condition", np);
+            condition = ConditionUtils.leafCondition(type, match, name, value);
```

Comment:
Semantically, I'm not following why a UserAttribute (which is a Condition) should need to go and get a Condition. Are we preparing for other condition types (e.g. a DeviceTypeCondition which would also implement Condition) that might want to share the same match types?